File: backend/models/warehouse.py
from sqlalchemy import Column, String, Integer, JSON
from sqlalchemy.dialects.postgresql import UUID
from database.db import Base
import uuid

class Warehouse(Base):
    __tablename__ = "warehouses"
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    dataset_id = Column(UUID(as_uuid=True), nullable=False)
    fact_table_name = Column(String)
    measures = Column(JSON)       # list of numeric columns treated as measures
    dimensions = Column(JSON)     # list of dimension table definitions
    data_dictionary = Column(JSON)  # column-level metadata

File: frontend/src/types/warehouse.ts
export interface Measure {
  column: string;
  type: string;
  aggregation: string;
}

export interface Dimension {
  column: string;
  dimension_table: string;
  distinct_values: number;
}

export interface DataDictionaryEntry {
  column: string;
  data_type: string;
  role: string;
  null_count: number;
  distinct_values: number;
}

export interface Warehouse {
  dataset_id: string;
  fact_table_name: string;
  measures: Measure[];
  dimensions: Dimension[];
  data_dictionary: DataDictionaryEntry[];
}

export interface WarehouseGenerateResponse extends Warehouse {
  warehouse_id: string;
}

export interface WarehouseData extends Warehouse {
  warehouse_id: string;
  measures: Measure[];
  dimensions: Dimension[];
}
